independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -578,20 +578,24 @@ private void validateBeans(List<Throwable> errors) {
                     // Disposers are not inherited
                     disposerMethods.add(method);
                 } else if (annotationStore.hasAnnotation(method, DotNames.OBSERVES)) {
-                    // TODO observers are inherited
```

Comment:
This shouldn't be removed - it's about discovering inherited observers.

Reply:
Not sure what you mean by that, but I can re-add that TODO